src/ipfs/encoded.ts

```diff
@@ -15,12 +17,33 @@ export const add = async (content: FileContent, key: Maybe<string>): Promise<Add
   // can't cbor encode blobs ie file streams
   const normalized = isBlob(content) ? await blob.toUint8Array(content) : content
   const encoded = cbor.encode(normalized)
-  const toAdd = isJust(key) ? await crypto.aes.encrypt(encoded, key) : encoded
+  
+  if (!isJust(key)) {
+    return basic.add(encoded)
+  }
+  
+  const alg = SymmAlg.AES_GCM
+  const cip = await crypto.aes.encrypt(encoded, key, alg)
+  const toAdd = cbor.encode({ alg, cip })
```

Comment:
The main attraction:
Instead of storing the encrypted blocks directly, we store a cbor block that contains the cipher in `cip` and the algorithm in `alg`.